host/frontend/host-orchestrator/instancemanager_test.go

```diff
@@ -65,6 +65,79 @@ func TestCreateCVDInvalidRequestsEmptyFields(t *testing.T) {
 	}
 }
 
+func TestCreateCVDNewOperationFails(t *testing.T) {
+	dir := t.TempDir()
+	opName := "operation-1"
+	om := NewMapOM(func() string { return opName })
+	fetchCVDHandler := NewFetchCVDHandler(dir, &FakeFetchCVDDownloader{t: t})
+	im := NewInstanceManager(fetchCVDHandler, om)
+	req := CreateCVDRequest{
+		BuildInfo: &BuildInfo{
+			BuildID: "1234",
+			Target:  "aosp_cf_x86_64_phone-userdebug",
+		},
+		FetchCVDBuildID: "1",
+	}
+
+	im.CreateCVD(req)
+	_, err := im.CreateCVD(req)
+
+	var appErr *AppError
+	if !errors.As(err, &appErr) {
+		t.Errorf("error type <<\"%T\">> not found in error chain", appErr)
+	}
+	var newOperationErr NewOperationError
+	if !errors.As(err, &newOperationErr) {
+		t.Errorf("error type <<\"%T\">> not found in error chain", newOperationErr)
+	}
+	im.launchCVDWG.Wait()
+	expectedOp := OperationData{
+		Name: opName,
+		Done: false,
+	}
+	op, _ := om.GetOperation(opName)
+	if op != expectedOp {
+		t.Errorf("expected <<%+v>>, got %+v", expectedOp, op)
+	}
+
+}
+
+func TestCreateCVDFetchCVDFails(t *testing.T) {
+	dir := t.TempDir()
+	opName := "operation-1"
+	om := NewMapOM(func() string { return opName })
+	fetchCVDHandler := NewFetchCVDHandler(dir, &AlwaysFailsFetchCVDDownloader{})
+	im := NewInstanceManager(fetchCVDHandler, om)
+	req := CreateCVDRequest{
+		BuildInfo: &BuildInfo{
+			BuildID: "1234",
+			Target:  "aosp_cf_x86_64_phone-userdebug",
+		},
+		FetchCVDBuildID: "1",
+	}
+	returnedOp := OperationData{
+		Name: opName,
+		Done: false,
+	}
+	completedOp := OperationData{
+		Name: opName,
+		Done: true,
+		Result: OperationResultData{
+			Error: OperationErrorData{"failed to download fetch_cvd"},
+		},
+	}
+
+	op, _ := im.CreateCVD(req)
+	if op != returnedOp {
+		t.Errorf("expected <<%+v>>, got %+v", returnedOp, op)
+	}
+	im.launchCVDWG.Wait()
+	op, _ = om.GetOperation(opName)
+	if op != completedOp {
```

Comment:
done comparing only the relevant fields. Operation stores the error message as string, which is going to be presented to the client as is, rather than a Golang error object therefore the plain string comparison.

Reply:
acknowledged, created and used a constant to avoid updating the test when the text changes.

Reply:
Even with the constant I don't see any value in checking the string, but at least it shouldn't hurt now.